test/assert.js

```diff
@@ -648,6 +652,17 @@ test('.throws() fails if passed a bad value', t => {
 	t.end();
 });
 
+test('promise .throws() fails when promise is resolved', t => {
+	return assertions.throws(Promise.resolve('foo'))
+	.catch(err => {
+		testFailResults(t, err, {
```

Comment:
Thanks for the tip and reviews!
I got it to work now :), but I had to pass the `lastFailure` to the `failsWith` function otherwise it would be assigned to `null`. Is this OK?